### DISCUSSION TITLE
Brightness

### DISCUSSION THREAD

**Discussion by westoneichner** [simpleavr/s14clock#17] [General]
Hello!

Any chance of making the display brighter than level 3? I see by using the flash animation the digits become brighter. Or is there any way to edit the firmware to eliminate multiplexing?

**Comment by simpleavr** [comment]
The multiplexing is already highly optimized over the development of this design.

The "flash" transition is actually the most recent animation added.

We can achieve to single out a single digit to be brighter than the others by stealing "on" cycles from the other digits. If we make longer the "on" cycles for all digits, we will loss proper multiplexing and we will see bright digits light up one by one, as the refresh rate is low.
The mutilplexing also requires proper on / off times as we also need to compensate the brightness on low segment (say a '-') and high segment (vs say a 'W' character) so they appears same brightness.

It is however possible to add logic to increase brightness when there are only a few (say 4 or 6) digits to display, as the blank digits can be skipped and we can use those cycles to turn on the 4 or 6 digits at a higher rate, thus achieve additional brightness.

As for the main code repository, there is no intention to add such feature. I had played w/ this design for more than 3 days and I feel there is very little to improve on the multiplexing.

The other way is to employ LED drivers (which I did in the original designs). That path is very different in that the project will be way more difficult to build (not as reliable as current simple design), and also limits the animation / transition options.

**Reply by westoneichner** [reply]
Thank you so much!
